test/lib/knock-knock.js

```diff
@@ -50,121 +34,164 @@ describe('lib/knock-knock', function () {
+        Fs.readFile.yields(null, out);
+        done();
+      });
+
+      it('executes all commands', function (done) {
+        KnockKnock({ key: 'value -v' }, function (err, results) {
           expect(err).to.be.null();
-          expect(results.env).to.equal('testenv');
+          expect(results.key).to.equal('value');
           done();
         });
       });
+    });
 
-      it('has a key called node', function (done) {
+    describe('when package.json is not found', function () {
+      it('yields an error', function (done) {
         KnockKnock(function (err, results) {
-          expect(err).to.be.null();
-          expect(results.node).to.equal('4.2.1');
+          expect(err).to.be.instanceof(Error);
+          expect(err.message).to.equal('readFile');
+          expect(results).to.be.undefined();
           done();
         });
       });
+    });
 
-      it('has a key called npm', function (done) {
+    describe('when executing a command fails', function () {
+      beforeEach(function (done) {
+        Child.exec.onFirstCall().yields(new Error('exec error').toString())
+                  .onSecondCall().yields(new Error('exec error').toString());
+        Fs.readFile.yields(null, out);
+        done();
+      });
+
+      it('stringifies the err', function (done) {
         KnockKnock(function (err, results) {
           expect(err).to.be.null();
-          expect(results.npm).to.equal('2.14.7');
+          expect(results.node).to.equal('Error: exec error');
+          expect(results.npm).to.equal('Error: exec error');
           done();
         });
       });
     });
-  });
-
-  describe('when options are passed in', function () {
-    beforeEach(function (done) {
-      Child.exec.onFirstCall().yields(null, '4.2.1')
-                .onSecondCall().yields(null, '2.14.7')
-                .onThirdCall().yields(null, 'value');
-      Fs.readFile.yields(null, out);
-      done();
-    });
 
-    it('executes all commands', function (done) {
-      KnockKnock({ key: 'value -v' }, function (err, results) {
-        expect(err).to.be.null();
-        expect(results.key).to.equal('value');
+    describe('when options is not an object', function () {
+      it('throws an error', function (done) {
+        expect(function () {
+          return KnockKnock('string', function () {
+            return;
+          });
+        }).to.throw();
         done();
       });
     });
   });
 
-  describe('when package.json is not found', function () {
-    it('yields an error', function (done) {
-      KnockKnock(function (err, results) {
-        expect(err).to.be.instanceof(Error);
-        expect(err.message).to.equal('readFile');
-        expect(results).to.be.undefined();
+  describe('handling newline characters', function () {
+    before(function (done) {
+      // eslint-disable-next-line handle-callback-err
+      Child.exec('npm -v', function (err, stdout) {
+        npm_version = stdout.replace(/\n/g, '');
```

Comment:
Just wanted to get it to work initially. If you have an idea on cleaning it up I'm all for it. 

I also did this to get the `npm version` since the process doesn't capture it without adding extra text.